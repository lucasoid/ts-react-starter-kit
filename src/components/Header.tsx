import * as React from 'react';
import { ErrorBoundary } from '~components/errors/ErrorBoundary';
import { MainMenu } from '~components/MainMenu';
import { Themes, ThemeContext } from '~theme';
import * as styles from './App.css';

interface IHeaderProps {
    title: string;
}

export const Header: React.FC<IHeaderProps> = props => {
    let { theme } = React.useContext(ThemeContext);

    return (
        <>
            <header className={`${styles.header} ${theme === Themes.DARK ? styles.dark : styles.light}`}>
                <h1 className={styles.title}>{props.title}</h1>
            </header>
            <ErrorBoundary>
                <MainMenu />
            </ErrorBoundary>
        </>
    );
};
