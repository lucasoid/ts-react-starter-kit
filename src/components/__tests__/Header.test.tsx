import * as React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../Header';
import { Themes } from '~theme';

jest.mock('react', () => {
    const original = jest.requireActual('react');
    return {
        __esModule: true,
        ...original,
        useContext: jest.fn(),
    };
});

describe('Header', () => {
    it('displays a themed header', () => {
        // @ts-ignore
        React.useContext.mockImplementationOnce(() => ({ theme: Themes.LIGHT }));
        let lightWrapper = shallow(<Header title="Welcome" />);
        expect(lightWrapper).toMatchSnapshot();

        // @ts-ignore
        React.useContext.mockImplementationOnce(() => ({ theme: Themes.DARK }));
        let darkWrapper = shallow(<Header title="Welcome" />);
        expect(darkWrapper).toMatchSnapshot();
    });
});
