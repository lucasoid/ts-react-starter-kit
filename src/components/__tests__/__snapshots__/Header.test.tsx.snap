// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header displays a themed header 1`] = `
<Fragment>
  <header
    className="header light"
  >
    <h1
      className="title"
    >
      Welcome
    </h1>
  </header>
  <ErrorBoundary>
    <MainMenu />
  </ErrorBoundary>
</Fragment>
`;

exports[`Header displays a themed header 2`] = `
<Fragment>
  <header
    className="header dark"
  >
    <h1
      className="title"
    >
      Welcome
    </h1>
  </header>
  <ErrorBoundary>
    <MainMenu />
  </ErrorBoundary>
</Fragment>
`;
